routes/account.js

```diff
@@ -102,6 +102,50 @@ function accountRoutes (server, options, next) {
     }
   }
 
+  var patchAccountRoute = {
+    method: 'PATCH',
+    path: '/session/account',
+    config: {
+      auth: false,
+      validate: {
+        headers: validations.bearerTokenHeader,
+        payload: validations.accountPayload,
+        failAction: joiFailAction
+      }
+    },
+    handler: function (request, reply) {
+      var sessionId = toBearerToken(request)
+      admins.validateSession(sessionId)
+      .then(function (doc) {
+        throw errors.FORBIDDEN_ADMIN_ACCOUNT
+      })
+      var newUsername = request.payload.data.attributes.username
+      var newPassword = request.payload.data.attributes.password
+
+      return sessions.find(sessionId)
+
+      .then(function (session) {
+        return accounts.update(session.account, {
+          username: newUsername,
+          password: newPassword
+        }, {
+          include: request.query.include
+        })
+      })
```

Comment:
we need to rearrange the code right now. Throwing an error on line 120 will not stop the code flow, only an "Unhandled error" would be logged to the console, the `return sessions.find(sessionId)` would be executed anyway. 

You can do it like in the `GET /session/account` handler:

```diff
- admins.validateSession(sessionId)
- .then(function (doc) {
-   throw errors.FORBIDDEN_ADMIN_ACCOUNT
- })
- var newUsername = request.payload.data.attributes.username
- var newPassword = request.payload.data.attributes.password
- 
- return sessions.find(sessionId)

+ var newUsername = request.payload.data.attributes.username
+ var newPassword = request.payload.data.attributes.password
+ admins.validateSession(sessionId)
+ 
+ .then(function (doc) {
+   throw errors.FORBIDDEN_ADMIN_ACCOUNT
+ })
+ 
+ .catch(function (error) {
+   if (error.name === 'not_found') {
+     return sessions.find(sessionId)
+   }
+ 
+   throw error
+ })
```

That way `session.find(sessionId)` only gets executed if session is not by an admin. Makes sense?